$DragonFly: src/gnu/usr.bin/gdb/libgdb/patches/Attic/i386fbsd-tdep.c.patch,v 1.2 2007/01/12 07:17:56 dillon Exp $

Index: i386fbsd-tdep.c
===================================================================
RCS file: /cvs/src/contrib/gdb-6.2.1/gdb/i386fbsd-tdep.c,v
retrieving revision 1.1
diff -u -r1.1 i386fbsd-tdep.c
--- i386fbsd-tdep.c	24 Oct 2004 19:56:25 -0000	1.1
+++ i386fbsd-tdep.c	12 Jan 2007 07:14:32 -0000
@@ -45,7 +45,7 @@
 CORE_ADDR i386fbsd_sigtramp_end_addr = 0xbfbfdff0;
 
 /* From <machine/signal.h>.  */
-static int i386fbsd_sc_reg_offset[] =
+int i386fbsd_sc_reg_offset[] =
 {
   8 + 14 * 4,			/* %eax */
   8 + 13 * 4,			/* %ecx */
@@ -77,7 +77,7 @@
      its FPU emulator in `struct fpreg'.  */
   tdep->gregset_reg_offset = i386fbsd_r_reg_offset;
   tdep->gregset_num_regs = ARRAY_SIZE (i386fbsd_r_reg_offset);
-  tdep->sizeof_gregset = 18 * 4;
+  tdep->sizeof_gregset = 20 * 4;
   tdep->sizeof_fpregset = 176;
 
   /* FreeBSD uses -freg-struct-return by default.  */
@@ -113,12 +113,12 @@
 /* From <machine/reg.h>.  */
 static int i386fbsd4_r_reg_offset[] =
 {
-  10 * 4, 9 * 4, 8 * 4, 7 * 4,	/* %eax, %ecx, %edx, %ebx */
-  16 * 4, 5 * 4,		/* %esp, %ebp */
-  4 * 4, 3 * 4,			/* %esi, %edi */
-  13 * 4, 15 * 4,		/* %eip, %eflags */
-  14 * 4, 17 * 4,		/* %cs, %ss */
-  2 * 4, 1 * 4, 0 * 4, 18 * 4	/* %ds, %es, %fs, %gs */
+  11 * 4, 10 * 4, 9 * 4, 8 * 4,	/* %eax, %ecx, %edx, %ebx */
+  18 * 4, 6 * 4,		/* %esp, %ebp */
+  5 * 4, 4 * 4,			/* %esi, %edi */
+  15 * 4, 17 * 4,		/* %eip, %eflags */
+  16 * 4, 19 * 4,		/* %cs, %ss */
+  3 * 4, 2 * 4, 1 * 4, 0 * 4	/* %ds, %es, %fs, %gs */
 };
 
 /* From <machine/signal.h>.  */
@@ -128,14 +128,14 @@
   20 + 10 * 4,			/* %ecx */
   20 + 9 * 4,			/* %edx */
   20 + 8 * 4,			/* %ebx */
-  20 + 17 * 4,			/* %esp */
+  20 + 18 * 4,			/* %esp */
   20 + 6 * 4,			/* %ebp */
   20 + 5 * 4,			/* %esi */
   20 + 4 * 4,			/* %edi */
-  20 + 14 * 4,			/* %eip */
-  20 + 16 * 4,			/* %eflags */
-  20 + 15 * 4,			/* %cs */
-  20 + 18 * 4,			/* %ss */
+  20 + 15 * 4,			/* %eip */
+  20 + 17 * 4,			/* %eflags */
+  20 + 16 * 4,			/* %cs */
+  20 + 19 * 4,			/* %ss */
   20 + 3 * 4,			/* %ds */
   20 + 2 * 4,			/* %es */
   20 + 1 * 4,			/* %fs */
@@ -154,7 +154,7 @@
   /* FreeBSD 4.0 introduced a new `struct reg'.  */
   tdep->gregset_reg_offset = i386fbsd4_r_reg_offset;
   tdep->gregset_num_regs = ARRAY_SIZE (i386fbsd4_r_reg_offset);
-  tdep->sizeof_gregset = 19 * 4;
+  tdep->sizeof_gregset = 20 * 4;
 
   /* FreeBSD 4.0 introduced a new `struct sigcontext'.  */
   tdep->sc_reg_offset = i386fbsd4_sc_reg_offset;
